jsk_pcl_ros/src/kinfu_nodelet.cpp

```diff
@@ -33,236 +32,547 @@
+          tf::StampedTransform tf_map_to_odom;
+          tf::transformEigenToTF(map_to_odom, tf_map_to_odom);
+          tf_map_to_odom.setRotation(tf_map_to_odom.getRotation().normalized());
+          tf_broadcaster_.sendTransform(
+            tf::StampedTransform(tf_map_to_odom, caminfo_msg->header.stamp, "map", fixed_frame_id_));
+        }
+        catch (tf::TransformException e)
         {
-          Eigen::Affine3f map = initial_camera_pose_ * K * odom_camera.inverse();
-          tf::Transform map_odom_transform;
-          tf::transformEigenToTF(map, map_odom_transform);
-          tf_broadcaster_.sendTransform(tf::StampedTransform(
-                                                             map_odom_transform,
-                                                             depth_image->header.stamp,
-                                                             parent_frame_id_,
-                                                             child_frame_id_));
+          NODELET_FATAL("%s", e.what());
         }
-        tf::Transform kinfu_origin;
-        Eigen::Affine3f inverse_camera_pose = camera_pose.inverse();
-        tf::transformEigenToTF(inverse_camera_pose, kinfu_origin);
-        tf_broadcaster_.sendTransform(tf::StampedTransform(
-                                                           kinfu_origin,
-                                                           depth_image->header.stamp,
-                                                           depth_image->header.frame_id,
-                                                           kinfu_origin_frame_id_));
-        Eigen::Affine3f camera_diff = K.inverse();
-        geometry_msgs::PoseStamped ros_camera_diff;
-        tf::poseEigenToMsg(camera_diff, ros_camera_diff.pose);
-        ros_camera_diff.header = depth_image->header;
-        pub_pose_.publish(ros_camera_diff);
-
-        // scene cloud.
-        pcl::gpu::DeviceArray<pcl::PointXYZ> extracted = kinfu_->volume().fetchCloud(cloud_buffer_device_);             
-        pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>);
-        pcl::PointCloud<pcl::PointXYZ>::Ptr transformed_cloud(new pcl::PointCloud<pcl::PointXYZ>);
-        extracted.download (cloud->points);
-        cloud->width = (int)cloud->points.size ();
-        cloud->height = 1;
-        //pcl::transformPointCloud(*cloud, *transformed_cloud, camera_pose.inverse());
-        sensor_msgs::PointCloud2 ros_cloud;
-        pcl::toROSMsg(*cloud, ros_cloud);
-        ros_cloud.header.stamp = depth_image->header.stamp;
-        ros_cloud.header.frame_id = kinfu_origin_frame_id_;
-        pub_cloud_.publish(ros_cloud);
       }
-      catch (...) {
-        NODELET_ERROR("Failed to lookup transform from %s to %s",
-                          child_frame_id_.c_str(),
-                          depth_image->header.frame_id.c_str());
-        return;
+    }
+
+    // publish generated depth image
```

Comment:
I think that generated depth in necessary only in limited use case. So how about adding rosparam to enalbe it and setting disabled as default?